go-controller/pkg/ovn/egressip.go

```diff
@@ -1911,10 +1899,54 @@ func (e *egressIPZoneController) deleteEgressIPStatusSetup(name string, status e
 				nextHopIP, name, e.GetNetworkScopedClusterRouterName(), err)
 		}
 	} else {
-		//FIXME: (mk) just nuke everything to do with this 'name'
-		klog.Errorf("Unable to get next hop IP and therefore there could be stale logical route policies for Egress IP %s", status.EgressIP)
+		// When no nextHopIP is found, This may happen when node object is already deleted.
+		// So compare validNextHopIPs associated with current eIP.Status and Nexthops present
+		// in the LogicalRouterPolicy, then delete nexthop(s) from LogicalRouterPolicy if
+		// it doesn't match with nexthops derived from eIP.Status.
+		policyPred := func(item *nbdb.LogicalRouterPolicy) bool {
```

Comment:
can you put all this into its own func - ensure only valid next hops

Reply:
done.